src/matzah.py

```diff
@@ -0,0 +1,102 @@
+import numpy as np
+import torch
+
+nn = torch.nn
+
+
+def adain(x, styleVec):
+    '''
+        This implements the AdaIN function described in the paper
+        
+        Paramaters
+        ---------
+        x: the intput
+        styleVec: is a vector that reperesents the desired style, this will 
+        initially start as a random vector somehow augmented with the choice
+        of character, it must have the same size as the number of feature maps
+        of x
+        Returns
+        -------
+        Adain'd feature map
+    '''
+    # this is elementwise multiplicationi
+    midPoint = styleVec.shape()[0] / 2
+    styleVecScale = styleVec[:midPoint]
+    styleVecBias = styleVec[midPoint:]
+    for i in range(x.shape()[0]):
+        feature_map = x[i, :, :, :]
+        x[i, :, :, :] = styleVecScale * (feature_map - torch.mean(
+            feature_map)) / torch.std(feature_map) + styleVecBias
+
+
+class StyleNetwork(nn.Module):
+    def __init__(self, fcSize, numFC, styleDim):
+        super(StyleNetwork, self).__init__()
+        self.fc1 = nn.linear(styleDim, fcSize)
+        self.fcLayers = [
+            nn.linear(fcSize, fcSize) for i in range(1, len(fcSizes))
+        ]
+        self.relu = nn.ReLU()
+
+    def forward(self, styleVec, classVec):
+        # concate the classVec and the StyleVec
+        style = torch.cat((styleVec, classVec), dim=0)
```

Comment:
This isn't right. We should concatenate the one-hot class vector to the output of the dense layers.

```python
x = nn.Linear(styleVec)
x = nn.Linear(x)
x = nn.Linear(x)
return torch.cat([x, classVec], dim=0)
```

Reply:
This is an easy fix, but what would be the practical difference? Would there be an issue with doing it this way.

I'm changing it right now, but just want to have the discussion